lib/hash_ring/ets.ex

```diff
@@ -371,11 +372,7 @@ defmodule ExHashRing.HashRing.ETS do
     :ok
   end
 
-  defp find_next_highest_item(_table, _ring_gen, 0, _hash) do
-    nil
-  end
-
-  defp find_next_highest_item(table, ring_gen, _num_nodes, hash) do
+  defp find_next_highest_item(table, ring_gen, num_nodes, hash) when num_nodes > 0 do
```

Comment:
Hmm, if we're going to guard against `num_nodes > 0` here, we should maintain this invariant all the way up. Can you add it to `find_node_inner` too?

Just so that it's immediately obvious to future readers that you cannot invoke `find_node_inner` with `num_nodes == 0`.

Reply:
That's a good idea! I just pushed up a new commit with that change. I also added the guard to `do_find_nodes/7`, does that sound reasonable to you?

```
--- a/lib/hash_ring/ets.ex
+++ b/lib/hash_ring/ets.ex
@@ -141,7 +141,7 @@ defmodule ExHashRing.HashRing.ETS do
     end
   end

-  defp find_node_inner(table, gen, num_nodes, key) do
+  defp find_node_inner(table, gen, num_nodes, key) when num_nodes > 0 do
     hash = Utils.hash(key)

     case find_next_highest_item(table, gen, num_nodes, hash) do
@@ -201,11 +201,13 @@ defmodule ExHashRing.HashRing.ETS do
     Enum.reverse(found)
   end

-  defp do_find_nodes(_table, _gen, num_nodes, _remaining, _hash, found, num_nodes) do
+  defp do_find_nodes(_table, _gen, num_nodes, _remaining, _hash, found, num_nodes)
+       when num_nodes > 0 do
     Enum.reverse(found)
   end

-  defp do_find_nodes(table, gen, num_nodes, remaining, hash, found, found_length) do
+  defp do_find_nodes(table, gen, num_nodes, remaining, hash, found, found_length)
+       when num_nodes > 0 do
     {number, node} = find_next_highest_item(table, gen, num_nodes, hash)

     if node in found do
```